### PR TITLE
CRD Version v1alpha2

### PR DESCRIPTION
**Pull request by jenniferubah**
With the recent changes to RHDH plugins, we are now enabling the notifications, notifications-backend and notificatinemail plugins and signals/signal backends plugins by default. Hence no longer allowing the admin to configure these values via the CR/CRD. As this is a breaking change, we introduce the v1alpha2 version of the CRD, remove these values from the CRD and `values.yaml` files. We also added the new custom plugins orchestrator.

- Updated rhdh-operator manifest using [this](https://github.com/batzionb/orchestrator-helm-operator/commit/78f089443e7bf993757d1fb5721d7d0e24c8aa6a) example

### CODE REVIEW

File: bundle/manifests/rhdh.redhat.com_orchestrators.yaml [hunk 1]
```diff
@@ -412,54 +380,30 @@ spec:
...
                     description: Orchestrator backend plugin information
                     properties:
                       integrity:
-                        default: sha512-Th5vmwyhHyhURwQo28++PPHTvxGSFScSHPJyofIdE5gTAb87ncyfyBkipSDq7fwj4L8CQTXa4YP6A2EkHW1npg==
+                        default: sha512-tS5cJGwjzP9esdTZvUFjw0O7+w9gGBI/+VvJrtqYJBDGXcEAq9iYixGk67ddQVW5eeUM7Tk1WqJlNj282aAWww==
                         description: Package SHA integrity
                         type: string
                       package:
-                        default: backstage-plugin-orchestrator-backend-dynamic@1.3.0
+                        default: backstage-plugin-orchestrator-backend-dynamic@1.4.0-rc.3
```
Comment (jordigilh) [inline bundle/manifests/rhdh.redhat.com_orchestrators.yaml]: I'd suggest to split the PR into 2:
* One PR that upgrades the CRD spec to the v1alpha2 with the changes we agreed (removing the plugins from the spec and changes requested by @jianrongzhang89 )
* One PR that updates the plugin versions in preparation for a new release.

File: config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@ metadata:
       managed resources (operands). To prevent data loss or disruption, you'll need
       to manually delete them.
     operators.openshift.io/valid-subscription: '["Red Hat Developer Hub"]'
-    repository: https://github.com/parodos-dev/orchestrator-helm-operator
+    repository: https://github.com/rhdhorchestrator/orchestrator-helm-operator
```
Comment (jordigilh) [inline config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml]: I would avoid making any updates not related to the spec changes. Besides, this is being covered in @pkliczewski 's renaming PR (#427 )

File: config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -71,13 +71,13 @@ spec:
   - Backstage
   links:
   - name: Orchestrator Operator
-    url: https://github.com/parodos-dev/orchestrator-helm-operator
+    url: https://github.com/rhdhorchestrator/orchestrator-helm-operator
   maintainers:
   - email: jgil@redhat.com
     name: Red Hat
-  maturity: alpha
+  maturity: stable
```
Comment (jordigilh) [inline config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml]: We're not yet GA, I would not think we're `stable` yet 😄  . We might be using the stable channel for the operator, but I consider that to be just a way to split between new development(fast/alpha) vs existing releases.

File: config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (jenniferubah) [reply config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml]: I had added this initially when making changes to this file and forgot to remove it after reviewing Piotr's PR. Will make the change.

File: config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (jenniferubah) [reply config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml]: Got it. I had read the field indicates the maturity level, then thought we were stable forgetting we are not GA yet. LOL. My bad. Will update it.

File: bundle/manifests/rhdh.redhat.com_orchestrators.yaml (hunk 1 above)
Comment (jenniferubah) [reply bundle/manifests/rhdh.redhat.com_orchestrators.yaml]: I thought @jianrongzhang89 already has a PR for this changes and would rebase after we merged this. I didn't know if he wanted those changes in this one. Will sync up with him now.

Yeah at the beginning, it was supposed to be just plugin changes and do rc-1 release afterwards. In hindsight, I should have split it into two PRs

Comment (jenniferubah) [reply bundle/manifests/rhdh.redhat.com_orchestrators.yaml:399]: I thought @jianrongzhang89 already has a PR for this changes and would rebase after we merged this. I didn't know if he wanted those changes in this one. Will sync up with him now.

Yeah at the beginning, it was supposed to be just plugin changes and do rc-1 release afterwards. In hindsight, I should have split it into two PRs

File: helm-charts/orchestrator/values.yaml [hunk 4]
```diff
@@ -56,27 +56,13 @@ rhdhPlugins: # RHDH plugins required for the Orchestrator
   npmRegistry: "https://npm.registry.redhat.com" # NPM registry is defined already in the container, but sometimes the registry need to be modified to use different versions of the plugin, for example: staging(https://npm.stage.registry.redhat.com) or development repositories
```
Comment (batzionb) [inline helm-charts/orchestrator/values.yaml]: @jenniferubah the -rc plugin versions are only available in the staging registry

File: config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (jenniferubah) [reply config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml]: Done

File: config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (jenniferubah) [reply config/manifests/bases/orchestrator-operator.clusterserviceversion.yaml]: Done

File: helm-charts/orchestrator/values.yaml (hunk 4 above)
Comment (jenniferubah) [reply helm-charts/orchestrator/values.yaml]: @batzionb, this is a default value and can be overridden via the CR. And QE can configure it for their testing.

Comment (batzionb) [reply helm-charts/orchestrator/values.yaml]: we want to have none functional default values in our main branch?

Comment (batzionb) [reply helm-charts/orchestrator/values.yaml]: @masayag WDYT?

File: bundle/manifests/rhdh.redhat.com_orchestrators.yaml (hunk 1 above)
Comment (jenniferubah) [reply bundle/manifests/rhdh.redhat.com_orchestrators.yaml]: Updated - removed the new plugin version changes

File: docs/main/existing-rhdh.md [hunk 5]
```diff
@@ -115,7 +115,7 @@ By default it should point to `http://sonataflow-platform-data-index-service.son
         pluginConfig:
           dynamicPlugins:
             frontend:
-              janus-idp.backstage-plugin-orchestrator:
+              red-hat-developer-hub.backstage-plugin-orchestrator:
```
Comment (jordigilh) [inline docs/main/existing-rhdh.md]: is this necessary for this PR?

File: helm-charts/orchestrator/templates/rhdh-operator.yaml [hunk 6]
```diff
@@ -218,26 +217,22 @@ data:
...
+        package: ./dynamic-plugins/dist/backstage-plugin-signals
         pluginConfig:
           dynamicPlugins:
             frontend:
-              redhat.plugin-signals: {}
+              backstage.plugin-signals: {}
       - disabled: false
-        package: "{{ .Values.rhdhPlugins.scope }}/{{ .Values.rhdhPlugins.notificationsBackend.package }}"
-        integrity: {{ .Values.rhdhPlugins.notificationsBackend.integrity }}
+        package: ./dynamic-plugins/dist/backstage-plugin-notifications-backend-dynamic
```
Comment (jordigilh) [inline helm-charts/orchestrator/templates/rhdh-operator.yaml:226]: I thought we were removing these 2 plugins... do we need to keep these references here?

File: docs/main/existing-rhdh.md (hunk 5 above)
Comment (batzionb) [reply docs/main/existing-rhdh.md]: if this PR updates the orchestrator plugins, it's necessary

File: helm-charts/orchestrator/templates/rhdh-operator.yaml (hunk 6 above)
Comment (batzionb) [reply helm-charts/orchestrator/templates/rhdh-operator.yaml:226]: if we won't have this code - the plugins won't appear, they are included in RHDH image but not enable by default at this point IIUC

Comment (jordigilh) [reply helm-charts/orchestrator/templates/rhdh-operator.yaml:226]: ack.

File: docs/main/existing-rhdh.md (hunk 5 above)
Comment (jordigilh) [reply docs/main/existing-rhdh.md]: This PR's goal is to make the necessary changes to the CRD spec to support `v1alpha2`. If the changes introduced from the changes from `v1alpha1` to `v1alpha2` require to upgrade these values then yes, they should go in. Otherwise then they should not be included. 
Another topic is that we must update these values in order for the orchestrator plugins to work, in which case it can be done in a different PR.

File: helm-charts/orchestrator/values.yaml (hunk 4 above)
Comment (jenniferubah) [reply helm-charts/orchestrator/values.yaml]: > we want to have none functional default values in our main branch?

If it's from a customer's perspective, I don't think they would have access to the staging registry right?
I had not changed during the last release and QE was able to test with it. So that's why I didn't need to change it as we prep for 1-4.

@jordigilh - do we usually change this while changing plugin version? I can change it in the next PR, if so.

File: docs/main/existing-rhdh.md (hunk 5 above)
Comment (jenniferubah) [reply docs/main/existing-rhdh.md]: Will remove it from this PR since we want a separate PR for the orchestrator plugin updates.

Comment (jenniferubah) [reply docs/main/existing-rhdh.md]: Done

File: helm-charts/orchestrator/values.yaml (hunk 4 above)
Comment (masayag) [reply helm-charts/orchestrator/values.yaml]: We must have https://github.com/rhdhorchestrator/orchestrator-helm-operator/pull/377  (bump RHDH to 1.4) before merging this PR, since only in RHDH-1.4 the notifications plugins are in-image.

as for Orchestrator plugins and NPM registry - we should maintain their compatibility and make sure that each change to the plugins matches the default npm registry (when release, it will be updated to https://npm.registry.redhat.com), but prior to it, it can only be installed internally. 

it isn't optimal since we won't be able to install it on demo clusters, unless we publish the plugins externally.

**Comment by pkliczewski** [comment]
@jordigilh please rereview this PR

File: helm-charts/orchestrator/templates/rhdh-operator.yaml [hunk 7]
```diff
@@ -202,12 +202,11 @@ data:
...
                     path: /orchestrator
       - disabled: false
-        package: "{{ .Values.rhdhPlugins.scope }}/{{ .Values.rhdhPlugins.notifications.package }}"
-        integrity: {{ .Values.rhdhPlugins.notifications.integrity }}
+        package: ./dynamic-plugins/dist/backstage-plugin-notifications
         pluginConfig:
           dynamicPlugins:
             frontend:
-              redhat.plugin-notifications:
+              backstage.plugin-notifications:
```
Comment (jordigilh) [inline helm-charts/orchestrator/templates/rhdh-operator.yaml]: is this change required for `v1alpha2`?

File: docs/main/existing-rhdh.md [hunk 8]
```diff
@@ -154,7 +154,7 @@ To include the Notification Plugin append this configuration to the ConfigMap:
         pluginConfig:
           dynamicPlugins:
             frontend:
-              redhat.plugin-signals: {}
+              backstage.plugin-signals: {}
```
Comment (jordigilh) [inline docs/main/existing-rhdh.md]: is this change required for `v1alpha2`?

File: docs/main/existing-rhdh.md [hunk 9]
```diff
@@ -138,7 +138,7 @@ To include the Notification Plugin append this configuration to the ConfigMap:
         pluginConfig:
           dynamicPlugins:
             frontend:
-              redhat.plugin-notifications:
+              backstage.plugin-notifications:
```
Comment (jordigilh) [inline docs/main/existing-rhdh.md]: is this change required for `v1alpha2`?

File: helm-charts/orchestrator/values.yaml (hunk 4 above)
Comment (jordigilh) [reply helm-charts/orchestrator/values.yaml]: Does it matter whether this PR is merged before or after? This branch (main) targets release-1.4/RHDH 1.4 anyway. When we change to reference the RHDH 1.4 operator channel should not deter us from merging any changes as long as they don't overlap or have dependencies to each other. What we should be aware is that the code is not ready to be tested until the other PR is merged to reflect the change in dependency versions.

 Or maybe I'm missing something...?

**Comment by jianrongzhang89** [comment]
@jenniferubah @pkliczewski @masayag @jordigilh why are we introducing version v1alpha2? Will it replace v1aplah1, or both version will co-exist? Do we have customers who already use v1alpha1, and if so how do they migrate to v1alpha2? Thanks.

**Comment by jordigilh** [comment]
> @jenniferubah @pkliczewski @masayag @jordigilh why are we introducing version v1alpha2? Will it replace v1aplah1, or both version will co-exist? Do we have customers who already use v1alpha1, and if so how do they migrate to v1alpha2? Thanks.

As discussed internally, we have agreed to raise the version due to the changes introduced here.

**Comment by jianrongzhang89** [comment]
> > @jenniferubah @pkliczewski @masayag @jordigilh why are we introducing version v1alpha2? Will it replace v1aplah1, or both version will co-exist? Do we have customers who already use v1alpha1, and if so how do they migrate to v1alpha2? Thanks.
> 
> As discussed internally, we have agreed to raise the version due to the changes introduced here.

Sounds good. Thank you @jordigilh

File: docs/main/existing-rhdh.md (hunk 8 above)
Comment (jenniferubah) [reply docs/main/existing-rhdh.md]: Not necessarily, will update it.

File: docs/main/existing-rhdh.md (hunk 9 above)
Comment (jenniferubah) [reply docs/main/existing-rhdh.md]: Not necessarily, will update it.

Comment (jenniferubah) [reply docs/main/existing-rhdh.md]: Done

File: docs/main/existing-rhdh.md (hunk 8 above)
Comment (jenniferubah) [reply docs/main/existing-rhdh.md]: Done

File: helm-charts/orchestrator/templates/rhdh-operator.yaml (hunk 7 above)
Comment (jenniferubah) [reply helm-charts/orchestrator/templates/rhdh-operator.yaml]: Updated as well.

File: helm-charts/orchestrator/values.yaml (hunk 4 above)
Comment (masayag) [reply helm-charts/orchestrator/values.yaml]: If this PR merged before, it will be broken, since in this PR we're still using RHDH 1.3.
Since this PR contains changes that were introduced only in RHDH-1.4, we need the other PR to be merged.
So with the other PR, we'll maintain integrity between PRs while with this one, this branch will be broken and will prevent from other developers to send/test PRs.

**Comment by pkliczewski** [comment]
@jordigilh do you think this change is ready? it is needed to release first RC build. it would be great to do it as soon as possible

**Review by jordigilh — APPROVED** [review]
/lgtm

File: helm-charts/orchestrator/values.yaml (hunk 4 above)
Comment (jenniferubah) [reply helm-charts/orchestrator/values.yaml]: Merged [RHDH 1.4](https://github.com/rhdhorchestrator/orchestrator-helm-operator/pull/437) changes first, before this one.
